cmake/BsoncxxUtil.cmake

```diff
@@ -10,17 +10,148 @@ function(bsoncxx_add_library TARGET OUTPUT_NAME LINK_TYPE)
         ${bsoncxx_sources}
     )
 
+    # Full ABI tag string to append to library output name.
+    # The value is determined at generator-time when using a multi-config generator.
+    # Otherwise, the value is determined at configure-time.
+    set(abi_tag "")
+
+    # ABI tag and properties.
+    if(1)
+        # Enforce ABI compatibility in dependent targets.
+        set_property(TARGET ${TARGET} APPEND PROPERTY COMPATIBLE_INTERFACE_STRING
+            BSONCXX_ABI_TAG_BUILD_TYPE
+            BSONCXX_ABI_TAG_MONGOC_LINK_TYPE
+            BSONCXX_ABI_TAG_POLYFILL_LIBRARY
+        )
+
+        # ABI version number. Only necessary for shared library targets.
+        if(LINK_TYPE STREQUAL "SHARED")
+            set(soversion _noabi)
+            set_target_properties(${TARGET} PROPERTIES SOVERSION ${soversion})
+            string(APPEND abi_tag "-v${soversion}")
+        endif()
+
+        # Build type:
+        # - 'd' for debug.
+        # - 'r' for release (including RelWithDebInfo and MinSizeRel).
+        if(1)
+            if(isMultiConfig)
```

Comment:
Good catch! Simplified as suggested.

This suggestion also revealed to me that `BSONCXX_ABI_TAG_BUILD_TYPE` as a compatible interface property is not well-motivated, given build configuration compatibility is better handled by CMake via `IMPORTED_CONFIGURATIONS` + its value being set to generator-expressions means it'd be computed according to downstream projects' configuration rather than the configuration of the installed libraries when they were built. Removed it from the list of compatible interface properties.